web/lib/queries/common/heroFields.ts

```diff
@@ -3,7 +3,8 @@ import linkSelectorFields from './actions/linkSelectorFields'
 export const heroFields = /* groq */ `{
     "type": coalesce(heroType, 'default'),
     "ratio": heroRatio,
-    "title": heroTitle,
+    "isBigTitle":isBigTitle,
+    "title": select(isBigTitle => select( heroType == 'fiftyFifty' => heroBigTitleFiftyFifty, heroBigTitleDefault) , heroTitle),
```

Comment:
this logic is wrong. it is checking for isBigTitle prior to the header type. Doing this implicates that if we toggle isBigTitle, and then switch to another header that doesn't have the bigTitle option, it will end up selecting the heroBigTitleDefault instead of heroTitle.

Reply:
and if you want to make it even safer, fetch isBigTitle only if headerType supports it.

"isBigTitle": if(supportsBigTitle) isBigTitle else false

Reply:
isBigTitle checkbox is shown only when the header supports it. Here if isBigTitle is false or undefined , heroTitle will be taken, if its true then relevant field value is selected.

Reply:
We should definitely have the refactor to avoid this nested select.

Reply:
> isBigTitle checkbox is shown only when the header supports it. Here if isBigTitle is false or undefined , heroTitle will be taken, if its true then relevant field value is selected.

You can choose 50/50 banner, mark the isBigTitle checkbox, and then switch to another banner. Then you have isBigTitle true without 5050 or defaultbanner being selected. isBigTitle is a document field, not a component field unfortunately #1952

Reply:
You are right, I overlooked this.